### ISSUE TITLE
Using quick-fix changes code

### ISSUE THREAD

**Issue by mindriven** [vimeo/psalm#9187]
As visualized below adding a suppression removes the assignment.
![psalm fix changing code](https://user-images.githubusercontent.com/4957654/214801960-94a401fd-c49c-4926-926a-519821bca147.gif)

**Comment by orklah** [comment]
This is not a Psalm's feature AFAIK (unfortunately 😄, that does looks cool). This should be reported to the plugin/IDE you use to integrate Psalm into your code.

@tm1000 is this some LSP plugin feature?

**Comment by tm1000** [comment]
@orklah technically this is the right place to report it because the code for the LSP lives here and not in any plugin/IDE.

**Comment by orklah** [comment]
Oh so the quick fix thing is integrated to Psalm? I wasn't aware of that!

**Comment by tm1000** [comment]
@orklah yeah I added it a while back: https://github.com/vimeo/psalm/blob/d7e9cbcef3c8d2eb56271e3e9d20664848be4d8b/src/Psalm/Internal/LanguageServer/Server/TextDocument.php#L382-L387

It'd be nice to get Psalter in there as well but its kinda impossible because Psalter runs in a different context. However being able to quickly suppress is nice :-)

**Comment by mindriven** [comment]
Nice to see the team devoting attention to reported issues!
This one is fairly small, impact on daily work is quite limited according to 1st comment from @tm1000 (I can confirm that is is really the case).
For me at this moment the blocker for adoption in the project is this issue: https://github.com/vimeo/psalm/issues/9197 Any chance of getting your help with this one?

**Comment by tm1000** [comment]
@mindriven looks like @weirdan has taken it on. At least for debug

**Comment by weirdan** [comment]
@mindriven do you still experience this issue?

**Comment by weirdan** [comment]
> This is because it's the last line in the file. If you add code below line 4 (like an echo) the suppression will work correctly.

Still happens for the code on the last line in file.
